.pre-commit-config.yaml

```diff
@@ -0,0 +1,40 @@
+repos:
+  - repo: meta
+    hooks:
+      - id: check-hooks-apply
+      - id: check-useless-excludes
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.3.0
+    hooks:
+      - id: check-yaml
+        exclude: 'pipeline/schema/.woodpecker/test-merge-map-and-sequence.yml'
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+  - repo: https://github.com/golangci/golangci-lint
+    rev: v1.55.1
+    hooks:
+      - id: golangci-lint
+  - repo: https://github.com/igorshubovych/markdownlint-cli
+    rev: v0.37.0
+    hooks:
+      - id: markdownlint-fix
+  - repo: https://github.com/mrtazz/checkmake
+    rev: 0.2.2
+    hooks:
+      - id: checkmake
```

Comment:
Needs to be installed locally or it will complain that it's not found. Overall users are forced to install these tools themselves, you can view them as "developer dependencies". 

If not (i.e. local hooks are not executed), precommit.ci should still do it/fail.

Reply:
This one is using `language: golang`, so go needs to be installed but not `checkmake` itself, `pre-commit` will prepare this environment automatically, see https://pre-commit.com/#golang